http/auth.go

```diff
@@ -51,31 +51,33 @@ func reCaptcha(host, secret, response string) (bool, error) {
 
 // authHandler processes the authentication for the user.
 func authHandler(c *fb.Context, w http.ResponseWriter, r *http.Request) (int, error) {
-	// NoAuth instances shouldn't call this method.
 	if c.NoAuth {
+		// NoAuth instances shouldn't call this method.
 		return 0, nil
-	}
-
-	// Receive the credentials from the request and unmarshal them.
-	var cred cred
-	if r.Body == nil {
-		return http.StatusForbidden, nil
-	}
-
-	err := json.NewDecoder(r.Body).Decode(&cred)
-	if err != nil {
-		return http.StatusForbidden, nil
-	}
+	} else if c.AuthMethod == "proxy" {
+		// Receive the Username from the Header.
+		cred.Username = r.Header.Get(c.LoginHeader)
```

Comment:
Can you merge the following lines here?

```
	// Checks if the user exists.
	u, err := c.Store.Users.GetByUsername(cred.Username, c.NewFS)
	if err != nil {
		return http.StatusForbidden, nil
    }
```

I.e.:

```
	// Receive the Username from the Header and check if it exists.
	u, err := c.Store.Users.GetByUsername(r.Header.Get(c.LoginHeader), c.NewFS)
	if err != nil {
		return http.StatusForbidden, nil
    }
```

This will allow not to evaluate `c.AuthMethod` again. See below.